### ISSUE TITLE
Node shutting itself down continously

### ISSUE THREAD

**Issue by reignmen** [pocketnetteam/pocketnet.core#91]
Node process quits with error:

2021-11-13T19:24:25Z RIDB rollback to block height 1438179 success!
2021-11-13T19:24:25Z 700004: Transaction RI data Not Found (b28886588f356077d95650660c3127bd3a3c4bfd16eddd95fcef10e60e87f560)
2021-11-13T19:24:25Z ERROR: Failed find reindexer transaction part (b28886588f356077d95650660c3127bd3a3c4bfd16eddd95fcef10e60e87f560)
2021-11-13T19:24:25Z ERROR: ConnectTip: ConnectBlock 0f928e498d3b6d1f10e432c390c3130ae988d2b9d38da439636c79ba78e7efb6 failed, failed-find-rtransaction (code 96)
2021-11-13T19:24:27Z 700004: Transaction RI data Not Found (ed979b6b69326b6cf639788d0ab05cf93bc7469686b11d6bad509484b2b2a4fc)
2021-11-13T19:24:27Z ERROR: Failed find reindexer transaction part (ed979b6b69326b6cf639788d0ab05cf93bc7469686b11d6bad509484b2b2a4fc)
2021-11-13T19:24:27Z ERROR: ConnectTip: ConnectBlock d3212bb21ca82699ad7916f42f12cf0a677d439e2760bd32bf285133597122d5 failed, failed-find-rtransaction (code 96)
2021-11-13T19:24:32Z Available coins count 0 BestHeader: 1438180 2eb80ccdc9243e1727c77bacdf4c542f19b861fda755db47433c14d97d699e53
2021-11-13T19:24:48Z Available coins count 0 BestHeader: 1438180 2eb80ccdc9243e1727c77bacdf4c542f19b861fda755db47433c14d97d699e53
2021-11-13T19:24:49Z RIDB rollback to block height 1438179 success!
2021-11-13T19:25:04Z Available coins count 0 BestHeader: 1438181 ea5422fee3c2267608dd93d9a0a92621b7659aa33643e44eae97ca2cd4c0a9d0
2021-11-13T19:25:06Z WARNING! Receive transaction, antibot check: 50 09191585a5d2e21d8d39203a7be15b0ecf0a631b8489a5fe6fee52dbc242da95
2021-11-13T19:25:20Z Available coins count 0 BestHeader: 1438181 ea5422fee3c2267608dd93d9a0a92621b7659aa33643e44eae97ca2cd4c0a9d0
2021-11-13T19:25:29Z RIDB rollback to block height 1438180 success!
2021-11-13T19:25:29Z ERROR: UndoReadFromDisk: Deserialize or I/O error 2 (26) - ReadVarInt(): size too large: iostream error
2021-11-13T19:25:29Z ERROR: DisconnectBlock(): failure reading undo data
2021-11-13T19:25:29Z ERROR: DisconnectTip(): DisconnectBlock 0f928e498d3b6d1f10e432c390c3130ae988d2b9d38da439636c79ba78e7efb6 failed
2021-11-13T19:25:29Z *** Failed to disconnect block; see debug.log for details
2021-11-13T19:25:29Z Error: Error: A fatal internal error occurred, see debug.log for details
Error: Error: A fatal internal error occurred, see debug.log for details
2021-11-13T19:25:29Z ERROR: ProcessNewBlock: ActivateBestChain failed (Failed to disconnect block; see debug.log for details (code 0))
2021-11-13T19:25:29Z ERROR: UndoReadFromDisk: Deserialize or I/O error 2 (26) - ReadVarInt(): size too large: iostream error
2021-11-13T19:25:29Z ERROR: DisconnectBlock(): failure reading undo data
2021-11-13T19:25:29Z ERROR: DisconnectTip(): DisconnectBlock 0f928e498d3b6d1f10e432c390c3130ae988d2b9d38da439636c79ba78e7efb6 failed
2021-11-13T19:25:29Z *** Failed to disconnect block; see debug.log for details
2021-11-13T19:25:29Z Error: Error: A fatal internal error occurred, see debug.log for details
Error: Error: A fatal internal error occurred, see debug.log for details
2021-11-13T19:25:29Z ERROR: ProcessNewBlock: ActivateBestChain failed (Failed to disconnect block; see debug.log for details (code 0))
2021-11-13T19:25:29Z ERROR: UndoReadFromDisk: Deserialize or I/O error 2 (26) - ReadVarInt(): size too large: iostream error
2021-11-13T19:25:29Z ERROR: DisconnectBlock(): failure reading undo data
2021-11-13T19:25:29Z ERROR: DisconnectTip(): DisconnectBlock 0f928e498d3b6d1f10e432c390c3130ae988d2b9d38da439636c79ba78e7efb6 failed
2021-11-13T19:25:29Z *** Failed to disconnect block; see debug.log for details
2021-11-13T19:25:29Z Error: Error: A fatal internal error occurred, see debug.log for details
Error: Error: A fatal internal error occurred, see debug.log for details
2021-11-13T19:25:29Z ERROR: ProcessNewBlock: ActivateBestChain failed (Failed to disconnect block; see debug.log for details (code 0))
2021-11-13T19:25:29Z tor: Thread interrupt
2021-11-13T19:25:29Z Shutdown: In progress...

**Desktop (please complete the following information):**
 - debian 11

after the shutdown involving error above, if i try to start pocketcoind i get the following output and shutown:
2021-11-14T08:19:36Z * Using 160.0MiB for in-memory UTXO set (plus up to 286.1MiB of unused mempool space)
2021-11-14T08:19:36Z init message: Loading block index...
2021-11-14T08:19:36Z Opening LevelDB in /home/pocketnet/.pocketcoin/blocks/index
2021-11-14T08:19:37Z Opened LevelDB successfully
2021-11-14T08:19:37Z Using obfuscation key for /home/pocketnet/.pocketcoin/blocks/index: 0000000000000000
2021-11-14T08:19:46Z LoadBlockIndexDB: last block file = 26
2021-11-14T08:19:46Z LoadBlockIndexDB: last block file info: CBlockFileInfo(blocks=7790, size=97655258, heights=1430743...1438182, time=2021-11-08...2021-11-13)
2021-11-14T08:19:46Z Checking all blk files are present...
2021-11-14T08:19:46Z Opening LevelDB in /home/pocketnet/.pocketcoin/chainstate
2021-11-14T08:19:47Z Opened LevelDB successfully
2021-11-14T08:19:47Z Using obfuscation key for /home/pocketnet/.pocketcoin/chainstate: 1fea1211fb56c2a2
2021-11-14T08:19:47Z Loaded best chain: hashBestChain=0f928e498d3b6d1f10e432c390c3130ae988d2b9d38da439636c79ba78e7efb6 height=1438180 date=2021-11-13T19:23:44Z progress=0.998854
2021-11-14T08:19:47Z init message: Rewinding blocks...
2021-11-14T08:19:50Z init message: Verifying blocks...
2021-11-14T08:19:50Z Check Reindexer DB... success.
2021-11-14T08:20:05Z Verifying last 6 blocks at level 3
2021-11-14T08:20:05Z [0%]...ERROR: UndoReadFromDisk: Deserialize or I/O error 2 (26) - ReadVarInt(): size too large: iostream error
2021-11-14T08:20:05Z ERROR: VerifyDB(): *** found bad undo data at 1438180, hash=0f928e498d3b6d1f10e432c390c3130ae988d2b9d38da439636c79ba78e7efb6

2021-11-14T08:20:05Z : Corrupted block database detected.
Please restart with -reindex or -reindex-chainstate to recover.
: Corrupted block database detected.
Please restart with -reindex or -reindex-chainstate to recover.
2021-11-14T08:20:05Z Aborted block database rebuild. Exiting.
2021-11-14T08:20:05Z Shutdown: In progress...
2021-11-14T08:20:05Z scheduler thread interrupt
2021-11-14T08:20:36Z Close reindexer DB
2021-11-14T08:20:36Z Shutdown: done

**Comment by nnmfnwl7** [comment]
Hit same same issue:

```
ERROR: ConnectTip: ConnectBlock d3212bb21ca82699ad7916f42f12cf0a677d439e2760bd32bf285133597122d5 failed, failed-find-rtransaction (code 96)
ERROR: UndoReadFromDisk: Deserialize or I/O error 2 (26) - ReadVarInt(): size too large: iostream error
ERROR: DisconnectBlock(): failure reading undo data
ERROR: DisconnectTip(): DisconnectBlock 2eb80ccdc9243e1727c77bacdf4c542f19b861fda755db47433c14d97d699e53 failed
*** Failed to disconnect block; see debug.log for details
Error: Error: A fatal internal error occurred, see debug.log for details
Error: Error: A fatal internal error occurred, see debug.log for details
ERROR: ProcessNewBlock: ActivateBestChain failed (Failed to disconnect block; see debug.log for details (code 0))
addcon thread exit
opencon thread exit
Shutdown: In progress...
msghand thread exit
net thread exit
Pocketcoin Staker terminated
scheduler thread interrupt
[wallet_pkoin] keypool return 1
Dumped mempool: 3e-06s to copy, 0.063258s to dump
Close reindexer DB
[wallet_pkoin] Releasing wallet
Shutdown: done
```

**Comment by Magic90007** [comment]
v0.20.7.0-6690586. Windows 11.

`2021-11-15T12:22:28Z txindex thread start
2021-11-15T12:22:28Z txindex is enabled
2021-11-15T12:22:28Z init message: Loading wallet...
2021-11-15T12:22:28Z txindex thread exit
2021-11-15T12:22:28Z Full reindexing database. This can take several days.
2021-11-15T12:22:28Z Deleting database indexes..
2021-11-15T12:22:28Z Rollback to first block..
2021-11-15T12:22:28Z Creating Sqlite database `main.sqlite3` structure..
2021-11-15T12:22:28Z Reindexing block file blk00000.dat...
2021-11-15T12:22:28Z mapBlockIndex.size() = 2
2021-11-15T12:22:28Z Best block: 00000fd0f6633d395541056e8adc32961e15f8133674b2e3937c4d210ced6f3f (0)
2021-11-15T12:22:28Z init message: Loading P2P addresses...
2021-11-15T12:22:28Z Loaded 0 addresses from peers.dat  0ms
2021-11-15T12:22:28Z init message: Loading banlist...
2021-11-15T12:22:28Z init message: Starting network threads...
2021-11-15T12:22:28Z net thread start
2021-11-15T12:22:28Z dnsseed thread start
2021-11-15T12:22:28Z addcon thread start
2021-11-15T12:22:28Z init message: Done loading
2021-11-15T12:22:28Z opencon thread start
2021-11-15T12:22:28Z msghand thread start
2021-11-15T12:22:28Z Loading addresses from DNS seeds (could take a while)
2021-11-15T12:22:28Z 0 addresses found from DNS seeds
2021-11-15T12:22:28Z dnsseed thread exit
2021-11-15T12:22:50Z Loaded 166008 blocks from external file in 22307ms
2021-11-15T12:22:50Z Reindexing block file blk00001.dat...
2021-11-15T12:23:03Z Loaded 90763 blocks from external file in 12637ms
2021-11-15T12:23:03Z Reindexing block file blk00002.dat...
2021-11-15T12:23:16Z Loaded 89057 blocks from external file in 12907ms
2021-11-15T12:23:16Z Reindexing block file blk00003.dat...
2021-11-15T12:23:27Z Loaded 71212 blocks from external file in 11216ms
2021-11-15T12:23:27Z Reindexing block file blk00004.dat...
2021-11-15T12:23:29Z Adding fixed seed nodes as DNS doesn't seem to be available.
2021-11-15T12:23:41Z Loaded 96394 blocks from external file in 14153ms
2021-11-15T12:23:41Z Reindexing block file blk00005.dat...
2021-11-15T12:23:59Z Loaded 123651 blocks from external file in 17583ms
2021-11-15T12:23:59Z Reindexing block file blk00006.dat...
2021-11-15T12:24:17Z Loaded 124077 blocks from external file in 17913ms
2021-11-15T12:24:17Z Reindexing block file blk00007.dat...
2021-11-15T12:24:34Z Loaded 115590 blocks from external file in 17386ms
2021-11-15T12:24:34Z Reindexing block file blk00008.dat...
2021-11-15T12:24:52Z Loaded 116218 blocks from external file in 17896ms
2021-11-15T12:24:52Z Reindexing block file blk00009.dat...
2021-11-15T12:25:00Z Loaded 43435 blocks from external file in 7935ms
2021-11-15T12:25:00Z Reindexing block file blk00010.dat...
2021-11-15T12:25:07Z Loaded 33741 blocks from external file in 6825ms
2021-11-15T12:25:07Z Reindexing block file blk00011.dat...
2021-11-15T12:25:13Z Loaded 30296 blocks from external file in 6319ms
2021-11-15T12:25:13Z Reindexing block file blk00012.dat...
2021-11-15T12:25:18Z Loaded 25186 blocks from external file in 4993ms
2021-11-15T12:25:18Z Reindexing block file blk00013.dat...
2021-11-15T12:25:25Z Loaded 34743 blocks from external file in 6577ms
2021-11-15T12:25:25Z Reindexing block file blk00014.dat...
2021-11-15T12:25:30Z Loaded 24408 blocks from external file in 4874ms
2021-11-15T12:25:30Z Reindexing block file blk00015.dat...
2021-11-15T12:25:35Z Loaded 34173 blocks from external file in 5858ms
2021-11-15T12:25:35Z Reindexing block file blk00016.dat...
2021-11-15T12:25:42Z Loaded 37399 blocks from external file in 6239ms
2021-11-15T12:25:42Z Reindexing block file blk00017.dat...
2021-11-15T12:25:48Z Loaded 38645 blocks from external file in 6036ms
2021-11-15T12:25:48Z Reindexing block file blk00018.dat...
2021-11-15T12:25:54Z Loaded 34423 blocks from external file in 6542ms
2021-11-15T12:25:54Z Reindexing block file blk00019.dat...
2021-11-15T12:26:00Z Loaded 27347 blocks from external file in 5889ms
2021-11-15T12:26:00Z Reindexing block file blk00020.dat...
2021-11-15T12:26:05Z Loaded 19080 blocks from external file in 4530ms
2021-11-15T12:26:05Z Reindexing block file blk00021.dat...
2021-11-15T12:26:09Z Loaded 14805 blocks from external file in 4201ms
2021-11-15T12:26:09Z Reindexing block file blk00022.dat...
2021-11-15T12:26:12Z Loaded 10944 blocks from external file in 3312ms
2021-11-15T12:26:12Z Reindexing block file blk00023.dat...
2021-11-15T12:26:15Z Loaded 9943 blocks from external file in 2859ms
2021-11-15T12:26:15Z Reindexing block file blk00024.dat...
2021-11-15T12:26:18Z Loaded 9638 blocks from external file in 2854ms
2021-11-15T12:26:18Z Reindexing block file blk00025.dat...
2021-11-15T12:26:21Z Loaded 10721 blocks from external file in 3278ms
2021-11-15T12:26:21Z Reindexing block file blk00026.dat...
2021-11-15T12:26:24Z Loaded 10020 blocks from external file in 3229ms
2021-11-15T12:26:24Z Reindexing finished
2021-11-15T12:27:33Z Error indexing social data: IndexRatings: Failed get score data for tx: 148f87e4728d19c4b7a2896bac3a45e69ac19b7a4bb4aa487e34672358f2e895

2021-11-15T12:27:33Z ERROR: ConnectTip: ConnectBlock ff88edd6ba74ed122e019e381e3e95fe696cb18b3094b8f478f9ef7caa4300b8 failed,  (code 0)
2021-11-15T12:27:33Z Failed to connect best block ( (code 0))
2021-11-15T12:27:33Z Shutdown: In progress...
2021-11-15T12:27:33Z opencon thread exit
2021-11-15T12:27:33Z net thread exit
2021-11-15T12:27:33Z addcon thread exit
2021-11-15T12:27:33Z msghand thread exit
2021-11-15T12:27:33Z scheduler thread interrupt
2021-11-15T12:27:44Z Error: Close: 5; Failed to close database main.sqlite3: database is locked
2021-11-15T12:27:44Z Shutdown: done`

**Comment by Magic90007** [comment]
> 
> 
> v0.20.7.0-6690586. Windows 11.
> 
> ```
> 2021-11-15T12:22:28Z txindex thread start
> 2021-11-15T12:22:28Z txindex is enabled
> 2021-11-15T12:22:28Z init message: Loading wallet...
> 2021-11-15T12:22:28Z txindex thread exit
> 2021-11-15T12:22:28Z Full reindexing database. This can take several days.
> 2021-11-15T12:22:28Z Deleting database indexes..
> 2021-11-15T12:22:28Z Rollback to first block..
> 2021-11-15T12:22:28Z Creating Sqlite database main.sqlite3 structure..
> 2021-11-15T12:22:28Z Reindexing block file blk00000.dat...
> 2021-11-15T12:22:28Z mapBlockIndex.size() = 2
> 2021-11-15T12:22:28Z Best block: 00000fd0f6633d395541056e8adc32961e15f8133674b2e3937c4d210ced6f3f (0)
> 2021-11-15T12:22:28Z init message: Loading P2P addresses...
> 2021-11-15T12:22:28Z Loaded 0 addresses from peers.dat  0ms
> 2021-11-15T12:22:28Z init message: Loading banlist...
> 2021-11-15T12:22:28Z init message: Starting network threads...
> 2021-11-15T12:22:28Z net thread start
> 2021-11-15T12:22:28Z dnsseed thread start
> 2021-11-15T12:22:28Z addcon thread start
> 2021-11-15T12:22:28Z init message: Done loading
> 2021-11-15T12:22:28Z opencon thread start
> 2021-11-15T12:22:28Z msghand thread start
> 2021-11-15T12:22:28Z Loading addresses from DNS seeds (could take a while)
> 2021-11-15T12:22:28Z 0 addresses found from DNS seeds
> 2021-11-15T12:22:28Z dnsseed thread exit
> 2021-11-15T12:22:50Z Loaded 166008 blocks from external file in 22307ms
> 2021-11-15T12:22:50Z Reindexing block file blk00001.dat...
> 2021-11-15T12:23:03Z Loaded 90763 blocks from external file in 12637ms
> 2021-11-15T12:23:03Z Reindexing block file blk00002.dat...
> 2021-11-15T12:23:16Z Loaded 89057 blocks from external file in 12907ms
> 2021-11-15T12:23:16Z Reindexing block file blk00003.dat...
> 2021-11-15T12:23:27Z Loaded 71212 blocks from external file in 11216ms
> 2021-11-15T12:23:27Z Reindexing block file blk00004.dat...
> 2021-11-15T12:23:29Z Adding fixed seed nodes as DNS doesn't seem to be available.
> 2021-11-15T12:23:41Z Loaded 96394 blocks from external file in 14153ms
> 2021-11-15T12:23:41Z Reindexing block file blk00005.dat...
> 2021-11-15T12:23:59Z Loaded 123651 blocks from external file in 17583ms
> 2021-11-15T12:23:59Z Reindexing block file blk00006.dat...
> 2021-11-15T12:24:17Z Loaded 124077 blocks from external file in 17913ms
> 2021-11-15T12:24:17Z Reindexing block file blk00007.dat...
> 2021-11-15T12:24:34Z Loaded 115590 blocks from external file in 17386ms
> 2021-11-15T12:24:34Z Reindexing block file blk00008.dat...
> 2021-11-15T12:24:52Z Loaded 116218 blocks from external file in 17896ms
> 2021-11-15T12:24:52Z Reindexing block file blk00009.dat...
> 2021-11-15T12:25:00Z Loaded 43435 blocks from external file in 7935ms
> 2021-11-15T12:25:00Z Reindexing block file blk00010.dat...
> 2021-11-15T12:25:07Z Loaded 33741 blocks from external file in 6825ms
> 2021-11-15T12:25:07Z Reindexing block file blk00011.dat...
> 2021-11-15T12:25:13Z Loaded 30296 blocks from external file in 6319ms
> 2021-11-15T12:25:13Z Reindexing block file blk00012.dat...
> 2021-11-15T12:25:18Z Loaded 25186 blocks from external file in 4993ms
> 2021-11-15T12:25:18Z Reindexing block file blk00013.dat...
> 2021-11-15T12:25:25Z Loaded 34743 blocks from external file in 6577ms
> 2021-11-15T12:25:25Z Reindexing block file blk00014.dat...
> 2021-11-15T12:25:30Z Loaded 24408 blocks from external file in 4874ms
> 2021-11-15T12:25:30Z Reindexing block file blk00015.dat...
> 2021-11-15T12:25:35Z Loaded 34173 blocks from external file in 5858ms
> 2021-11-15T12:25:35Z Reindexing block file blk00016.dat...
> 2021-11-15T12:25:42Z Loaded 37399 blocks from external file in 6239ms
> 2021-11-15T12:25:42Z Reindexing block file blk00017.dat...
> 2021-11-15T12:25:48Z Loaded 38645 blocks from external file in 6036ms
> 2021-11-15T12:25:48Z Reindexing block file blk00018.dat...
> 2021-11-15T12:25:54Z Loaded 34423 blocks from external file in 6542ms
> 2021-11-15T12:25:54Z Reindexing block file blk00019.dat...
> 2021-11-15T12:26:00Z Loaded 27347 blocks from external file in 5889ms
> 2021-11-15T12:26:00Z Reindexing block file blk00020.dat...
> 2021-11-15T12:26:05Z Loaded 19080 blocks from external file in 4530ms
> 2021-11-15T12:26:05Z Reindexing block file blk00021.dat...
> 2021-11-15T12:26:09Z Loaded 14805 blocks from external file in 4201ms
> 2021-11-15T12:26:09Z Reindexing block file blk00022.dat...
> 2021-11-15T12:26:12Z Loaded 10944 blocks from external file in 3312ms
> 2021-11-15T12:26:12Z Reindexing block file blk00023.dat...
> 2021-11-15T12:26:15Z Loaded 9943 blocks from external file in 2859ms
> 2021-11-15T12:26:15Z Reindexing block file blk00024.dat...
> 2021-11-15T12:26:18Z Loaded 9638 blocks from external file in 2854ms
> 2021-11-15T12:26:18Z Reindexing block file blk00025.dat...
> 2021-11-15T12:26:21Z Loaded 10721 blocks from external file in 3278ms
> 2021-11-15T12:26:21Z Reindexing block file blk00026.dat...
> 2021-11-15T12:26:24Z Loaded 10020 blocks from external file in 3229ms
> 2021-11-15T12:26:24Z Reindexing finished
> 2021-11-15T12:27:33Z Error indexing social data: IndexRatings: Failed get score data for tx: 148f87e4728d19c4b7a2896bac3a45e69ac19b7a4bb4aa487e34672358f2e895
> 
> 2021-11-15T12:27:33Z ERROR: ConnectTip: ConnectBlock ff88edd6ba74ed122e019e381e3e95fe696cb18b3094b8f478f9ef7caa4300b8 failed,  (code 0)
> 2021-11-15T12:27:33Z Failed to connect best block ( (code 0))
> 2021-11-15T12:27:33Z Shutdown: In progress...
> 2021-11-15T12:27:33Z opencon thread exit
> 2021-11-15T12:27:33Z net thread exit
> 2021-11-15T12:27:33Z addcon thread exit
> 2021-11-15T12:27:33Z msghand thread exit
> 2021-11-15T12:27:33Z scheduler thread interrupt
> 2021-11-15T12:27:44Z Error: Close: 5; Failed to close database main.sqlite3: database is locked
> 2021-11-15T12:27:44Z Shutdown: done
> ```

Update helps me

**Comment by andyoknen** [comment]
The synchronization issue has been fixed in version 0.20.*, which is in pre-release state and is being prepared for release.
I will close this issue
